### PR TITLE
feat:Added bookmarks in Schedule Page

### LINKED ISSUE #1660: Bookmark option in sessions list, Schedule fragment
**Actual Behaviour**

Just to enhance the user experience, I think we should add bookmark option in the list itself, apart from that in the sessions detail activity. 
This will also help maintain uniformity in the app, as in other fragments this feature is present. 
Also, the bookmark icon is not uniform through the app. I propose to use one common icon throughout the app.
**Expected Behaviour**

- [ ] Add bookmark option in session list under Schedule fragment.

- [ ] Make a common bookmark option for throughout the app.

**Steps to reproduce it**

Add steps to reproduce bugs or add information on the place where the feature should be implemented. Add links to a sample deployment or code.

**Screenshots of the issue**

device-2017-06-05-133852

device-2017-06-05-133820

**Would you like to work on the issue?**

Yes.

**Target Milestone to solve the issue**

Till which [milestone](https://github.com/fossasia/open-event-android/milestones) will you be fixing the issue? 
https://github.com/fossasia/open-event-android/milestone/28

### PR DESCRIPTION
**Pull request by shreyasnbhat**
Fixes #1660 

Screenshots are in the Issue comments itself.

### CODE REVIEW

**Comment by shreyasnbhat** [comment]
@the-dagger I am yet to do the thing you told me to do. Will do it soon and update the PR.

**Comment by shreyasnbhat** [comment]
@iamareebjamal @arp95 @mahikaw @the-dagger  Please review.

File: android/app/src/main/res/layout/item_schedule.xml [hunk 1]
```diff
@@ -105,6 +106,21 @@
...
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:layout_gravity="end"
+                android:layout_alignParentEnd="true"
+                android:layout_alignParentRight="true"
+                android:layout_alignParentBottom="true"
+                android:tint="@color/black"
+                android:background="?attr/selectableItemBackgroundBorderless"
+                android:contentDescription="@string/session_bookmark_status"
+                android:padding="6dp"
```
Comment (arp95) [inline android/app/src/main/res/layout/item_schedule.xml]: please use value from dimens.xml

File: android/app/src/main/java/org/fossasia/openevent/adapters/DayScheduleAdapter.java [hunk 2]
```diff
@@ -99,6 +105,45 @@ public void onBindViewHolder(DayScheduleViewHolder holder, int position) {
             holder.slotTrack.getBackground().setColorFilter(storedColor, PorterDuff.Mode.SRC_ATOP);
             holder.slotTrack.setText(sessionTrack.getName());
 
+            if(currentSession.isBookmarked()) {
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/adapters/DayScheduleAdapter.java]: I think the same functionality of this block is present elsewhere, please share the code between two.

It becomes unmaintainable this way

Comment (shreyasnbhat) [inline android/app/src/main/java/org/fossasia/openevent/adapters/DayScheduleAdapter.java]: @iamareebjamal Will do that.

Comment (shreyasnbhat) [inline android/app/src/main/java/org/fossasia/openevent/adapters/DayScheduleAdapter.java]: So what do you think I should do. I am setting the color for the bookmark icon in the schedule page. Do you think I should make this like a general all over the app?

Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/adapters/DayScheduleAdapter.java]: I'm talking about this code 

https://github.com/shreyasnbhat/open-event-android/blob/f1e4b5856cc3402879eb6c1dce85dabe9f3a4e46/android/app/src/main/java/org/fossasia/openevent/adapters/DayScheduleAdapter.java#L108-L145

It repeats a lot

Comment (shreyasnbhat) [inline android/app/src/main/java/org/fossasia/openevent/adapters/DayScheduleAdapter.java]: @iamareebjamal I'll try to generalize it a bit.

Comment (shreyasnbhat) [inline android/app/src/main/java/org/fossasia/openevent/adapters/DayScheduleAdapter.java]: @iamareebjamal I can do this refactor only after my PR related to global-search gets merged. As I need `ViewHolder` refactor to make this change.

Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/adapters/DayScheduleAdapter.java]: Well, please create a new issue for this then

Comment (shreyasnbhat) [inline android/app/src/main/java/org/fossasia/openevent/adapters/DayScheduleAdapter.java]: Yeah Sure. So I guess this could be merged now.

Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/adapters/DayScheduleAdapter.java]: Issues are to be created as encountered, or we may forget about them

**Comment by shreyasnbhat** [comment]
But if the code doesn't contain the issue currently how does that make
sense to create an issue?

On Jun 10, 2017 10:27 AM, "Areeb Jamal" <notifications@github.com> wrote:

*@iamareebjamal* commented on this pull request.
------------------------------

In android/app/src/main/java/org/fossasia/openevent/adapters/
DayScheduleAdapter.java
<https://github.com/fossasia/open-event-android/pull/1685#discussion_r121249698>
:

> @@ -99,6 +105,45 @@ public void onBindViewHolder(DayScheduleViewHolder holder, int position) {
             holder.slotTrack.getBackground().setColorFilter(storedColor,
PorterDuff.Mode.SRC_ATOP);
             holder.slotTrack.setText(sessionTrack.getName());

+            if(currentSession.isBookmarked()) {

Issues are to be created as encountered, or we may forget about them

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/fossasia/open-event-android/pull/1685#discussion_r121249698>,
or mute the thread

File: android/app/src/main/java/org/fossasia/openevent/adapters/DayScheduleAdapter.java (hunk 2 above)
Comment (shreyasnbhat) [inline android/app/src/main/java/org/fossasia/openevent/adapters/DayScheduleAdapter.java]: @iamareebjamal I will create the Issue as soon as it gets merged as the Issue doesn't exist in the codebase currently.

**Comment by shreyasnbhat** [comment]
@arp95 @iamareebjamal Please review. I'll refactor the code duplication in an Issue I'll be creating shortly but is dependent upon the #1565 getting merged as it involves `ViewHolder` refactoring.

**Comment by mahikaw** [comment]
@shreyasnbhat the second part of the issue about uniform bookmark icons has not been implemented. Please check the issue in detail for future, before addressing it in a PR. Could you update it now? Or shall I send a separate PR for it?

**Comment by shreyasnbhat** [comment]
@mahikaw What do you mean by uniform bookmark icon? You are talking about the icon which appears in the SessionDetailActivity?

**Comment by arp95** [comment]
i think shes talking about the one in schedule fragment.

**Comment by shreyasnbhat** [comment]
@arp95  Isn't the icon the same? Seems pretty uniform to me.

**Comment by mahikaw** [comment]
Yeah, we are using different bookmark icons at different places. We need to use one common icon. @shreyasnbhat

**Comment by shreyasnbhat** [comment]
@mahikaw  Could you tell me where are we using different icons? The bookmarks in the HomePage and the Schedule use the same icon.

**Comment by mahikaw** [comment]
Session details @shreyasnbhat

**Comment by shreyasnbhat** [comment]
@iamareebjamal Is this needed? If yes @mahikaw Do send a separate PR.

**Comment by arp95** [comment]
even i dont need its necessary. anyways go ahead @mahikaw

**Comment by mahikaw** [comment]
2 different icons are creating non-uniformity and occupying more space. This is trivial, yes but worth fixing I believe. That's the reason I had combined it with the main issue. @arp95 @shreyasnbhat

**Pull request merged**